cpp/include/patterns/copy_v_transform_reduce_nbr.cuh

```diff
@@ -0,0 +1,293 @@
+        GraphViewType::is_adj_matrix_transposed ? static_cast<vertex_t>(idx) : minor_offset;
+      auto e_op_result = evaluate_edge_op<GraphViewType,
+                                          AdjMatrixRowValueInputIterator,
+                                          AdjMatrixColValueInputIterator,
+                                          EdgeOp>()
+                           .compute(row,
+                                    col,
+                                    weight,
+                                    *(adj_matrix_row_value_input_first + row_offset),
+                                    *(adj_matrix_col_value_input_first + col_offset),
+                                    e_op);
+      if (update_major) {
+        accumulate_edge_op_result<update_major>(e_op_result_sum, e_op_result);
+      } else {
+        accumulate_edge_op_result<update_major>(*(result_value_output_first + minor_offset),
+                                                e_op_result);
+      }
+    }
+    if (update_major) { *(result_value_output_first + idx) = e_op_result_sum; }
+    idx += gridDim.x * blockDim.x;
+  }
+}
+
+}  // namespace detail
+
+/**
+ * @brief Iterate over the incoming edges to update vertex properties.
+ *
+ * This function is inspired by thrust::transfrom_reduce() (iteration over the incoming edges part)
+ * and thrust::copy() (update vertex properties part, take transform_reduce output as copy input).
+ *
+ * @tparam GraphViewType Type of the passed non-owning graph object.
+ * @tparam AdjMatrixRowValueInputIterator Type of the iterator for graph adjacency matrix row
+ * input properties.
+ * @tparam AdjMatrixColValueInputIterator Type of the iterator for graph adjacency matrix column
+ * input properties.
+ * @tparam EdgeOp Type of the binary (or ternary) edge operator.
+ * @tparam T Type of the initial value for reduction over the incoming edges.
+ * @tparam VertexValueOutputIterator Type of the iterator for vertex output property variables.
+ * @param handle RAFT handle object to encapsulate resources (e.g. CUDA stream, communicator, and
+ * handles to various CUDA libraries) to run graph algorithms.
+ * @param graph_view Non-owning graph object.
+ * @param adj_matrix_row_value_input_first Iterator pointing to the adjacency matrix row input
+ * properties for the first (inclusive) row (assigned to this process in multi-GPU).
+ * `adj_matrix_row_value_input_last` (exclusive) is deduced as @p adj_matrix_row_value_input_first +
+ * @p graph_view.get_number_of_adj_matrix_local_rows().
+ * @param adj_matrix_col_value_input_first Iterator pointing to the adjacency matrix column input
+ * properties for the first (inclusive) column (assigned to this process in multi-GPU).
+ * `adj_matrix_col_value_output_last` (exclusive) is deduced as @p adj_matrix_col_value_output_first
+ * + @p graph_view.get_number_of_adj_matrix_local_cols().
+ * @param e_op Binary (or ternary) operator takes *(@p adj_matrix_row_value_input_first + i), *(@p
```

Comment:
Seems to really need 4 or 5 parameters:  src, dest, optional weight, src_val, dst_val.

Reply:
Yeah... initially, this was designed without considering hypergraph partitioning.

In that case, we can provide src & dst as part of adj_matrix_row_value_input_first & adj_matrix_col_value_input_first with thrust counting iterator.

But if hypergrpah partitioning is considered, doing this from the user space requires something more complex (that increase the cost to access src (row-major) or dst (col-major) requiring binary search).

So, I fixed this recently but forget to update the comment :-) Thank you for catching this.

See the code below for a use case.
https://github.com/rapidsai/cugraph/pull/838/files#diff-615b093890b957db6e03f2a3f49d096bR243
```
    copy_v_transform_reduce_in_nbr(
      handle,
      pull_graph_view,
      adj_matrix_row_pageranks.begin(),
      thrust::make_constant_iterator(0) /* dummy */,
      [alpha] __device__(vertex_t src, vertex_t dst, weight_t w, auto src_val, auto dst_val) {
        return src_val * w * alpha;
      },
      unvarying_part,
      pageranks);
```

Reply:
Fixed.